spdx-utils/build.gradle

```diff
@@ -151,12 +151,24 @@ def generateEnumClass(String taskName, String description, String jsonUrl, Strin
 def generateLicenseTextResources(String description, Map<String, Map<String, String>> ids, String resourcePath) {
     logger.quiet("Determining SPDX $description texts...")
 
+    def scanCodeLicensePath = "$buildDir/SvnExport/licenses/scancode-toolkit"
+    def spdxIdToScanCodeKey = [:]
+
+    file(scanCodeLicensePath).listFiles({ it.name.endsWith('.yml') } as FileFilter).each { file ->
```

Comment:
A little more defensive option would be to list all *.LICENSE files and then for each check for the existance of a yaml file with the same name.  This would ensure that for each entry on the map there exists a .LICENSE file and thus ensure that line 185 reads an existing file.

Reply:
That's only half-true: It's correct that we'd have an existing file for each key in the map then, but that does not mean that line 185 always reads an existing file. That's because `id` could be a string that is not a key in the map, resulting in `null` to be returned from the map lookup and in a file named "null.LICENSE" to be read. As that file does not exist, that triggers the (expected) `FileNotFoundException` to try the remaining candidates.

I agree it's a bit ugly to let the code run into "null.LICENSE" being read instead of checking for that case earlier, but on the other hand it keeps the amount of code a bit smaller as we anyway expect the `FileNotFoundException` to (sometimes) be thrown. I felt this was good enough for a solution right now until we move this code to Kotlin script 😉

Reply:
Agree